### PR TITLE
[#10362] Allow receiving old messages after rejoining a pubchat

### LINKED ISSUE #10362: Can't load any messages after rejoining a public group
# Bug Report

## Problem

If I delete a public group and join it again, no messages will load as if it's an empty group. "Fetch more messages" doesn't work. 

#### Expected behavior

Previous messages are loaded.

#### Actual behavior

Nothing appears and "fetch more messages" button won't get it work. 
### Reproduction

- Join a public group 
- delete chat 
- join the same group again
### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: 1.2.0
[comment]: # (Android or iOS.)
- Operating System: Android Q

### PR DESCRIPTION
**Pull request by rasom**
fix #10362 

status: ready

### CODE REVIEW

File: src/status_im/chat/models/message.cljs [hunk 1]
```diff
@@ -107,9 +107,11 @@
 
 (defn- earlier-than-deleted-at?
   [{:keys [db]} {:keys [chat-id clock-value]}]
-  (let [{:keys [deleted-at-clock-value]}
+  (let [{:keys [deleted-at-clock-value public?]}
```
Comment (vkjr) [inline src/status_im/chat/models/message.cljs]: Maybe we can leave only one `earlier-than-deleted-at?` function since they are identical?

Comment (rasom) [reply src/status_im/chat/models/message.cljs]: yep we definitely can

Comment (rasom) [reply src/status_im/chat/models/message.cljs]: fixed

**Review by cammellos — APPROVED** [review]

**Comment by rasom** [comment]
this PR breaks "clear history" feature in pubchat

**Comment by qoqobolo** [comment]
### ISSUE 1: Can't fetch old messages after rejoining a public chat until relogin/opening Profile tab
Although the last message is displayed in the preview on the home screen after rejoining. 

#### Expected behavior: 
Can see old messages right after rejoining a public chat without relogin/tab switching.

_Steps:_
- Join public chat
- Delete public chat
- Rejoin public chat
- Tap Fetch more messages
- Navigate to Profile tab or relogin
- Navigate to Chat view

Android, iOS

ezgif com-gif-maker (2)

**Comment by rasom** [comment]
@qoqobolo issue 1 is fixed but this pr breaks "clear history" e2e test, so it can't be merged as is

**Comment by churik** [comment]
@rasom last commit can't be built

**Comment by rasom** [comment]
@churik I hope the next commit will be built successfully

**Pull request merged**